wandb/sdk/artifacts/artifact.py

```diff
@@ -1647,12 +1647,16 @@ def download(
 
         Raises:
             ArtifactNotLoggedError: If the artifact is not logged.
+            RuntimeError: If the artifact is attempted to be downloaded in offline mode.
         """
         self._ensure_logged("download")
 
         root = FilePathStr(str(root or self._default_root()))
         self._add_download_root(root)
 
+        if env.is_offline() or util._is_offline():
+            raise RuntimeError("Cannot download artifacts in offline mode.")
```

Comment:
`util._is_offline()` i don't think it is a good idea to use this function, as it has a side effect of calling `wandb.setup`

Reply:
What would be the best way to check `offline` mode coming from env var/cli/run? Is there a single function I can use?

Reply:
I'm reverting to `util._is_offline()` because it checks for the `run` to be available, and internally uses `wandb.run.settings`, unless I get a better candidate for this check.

Reply:
Added a TODO comment to fix this check to use a better way, since this is an anti-pattern @kptkin